pkg/statistics/region_collection.go

```diff
@@ -118,8 +138,8 @@ func (r *RegionStatistics) GetRegionStatsByType(typ RegionStatisticType) []*core
 	r.RLock()
 	defer r.RUnlock()
 	res := make([]*core.RegionInfo, 0, len(r.stats[typ]))
-	for _, r := range r.stats[typ] {
-		res = append(res, r.RegionInfo.Clone())
+	for regionID := range r.stats[typ] {
+		res = append(res, r.core.GetRegion(regionID).Clone())
```

Comment:
Do you need to check the region status that satisfy the required type? 
for example , some region  is down peer in the stats , but is healthy peer in th core.

Reply:
I think it's unnecessary since `(*RegionStatistics).Observe` will be called right after the region info is updated in `*RegionsInfo`, therefore according to the current code, there will be no inconsistency.

Reply:
What if we have some stores that are offline and each of them has many regions?